DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Linq;
+using System.Web.UI.WebControls;
+using DotNetNuke.Abstractions;
+using DotNetNuke.Common;
+using DotNetNuke.Common.Utilities;
+using DotNetNuke.Entities.Modules;
+using DotNetNuke.Services.FileSystem;
+using DotNetNuke.Services.FileSystem.Internal;
+using DotNetNuke.Services.Localization;
+using DotNetNuke.UI.Skins.Controls;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Dnn.Modules.ResourceManager
+{
+    public partial class EditFolderMapping : PortalModuleBase
+    {
+        private readonly INavigationManager _navigationManager;
+
+        private readonly IFolderMappingController _folderMappingController = FolderMappingController.Instance;
+        private int _folderMappingID = Null.NullInteger;
+
+        public EditFolderMapping()
+        {
+            this._navigationManager = this.DependencyProvider.GetRequiredService<INavigationManager>();
+        }
+
+        public int FolderPortalID
+        {
+            get
+            {
+                return this.IsHostMenu ? Null.NullInteger : this.PortalId;
+            }
+        }
+
+        public int FolderMappingID
+        {
+            get
+            {
+                if (this._folderMappingID == Null.NullInteger)
+                {
+                    if (!string.IsNullOrEmpty(this.Request.QueryString["ItemID"]))
+                    {
+                        int.TryParse(this.Request.QueryString["ItemID"], out this._folderMappingID);
+                    }
+                }
+
+                return this._folderMappingID;
+            }
+        }
+
+        protected override void OnInit(EventArgs e)
+        {
+            base.OnInit(e);
+
+            if (!this.UserInfo.IsSuperUser && !this.UserInfo.IsInRole(this.PortalSettings.AdministratorRoleName))
+            {
+                this.Response.Redirect(Globals.AccessDeniedURL(), true);
```

Comment:
Could this be added to the `NavigationManager`?

Reply:
I am not sure what you mean by this ? @ahoefling

Reply:
The Dnn `NavigationManager` manages several common routing scenarios in DNN. It doesn't include all of them, but is at a great starting point. The `Globals.AccessDeniedURL()` really should be ported into the `NavigationManager`. 

I propose the following changes
- Add new API to `INavigationManager` - `AccessDeniedUrl()`
- Implement new API in `NavigationManager` by porting the `Globals.AccessDeniedURL()` code
- Update the `Globals.AccessDeniedURL()` to use the `INavigationManager` new API

This may be a bit to refactor, it is sometimes hard to tell just by looking at the API structure. @bdukes or @mitchelsellers do you have any thoughts on this.

Reply:
Ha, I understand now :)

Reply:
Yeah, moving more navigation methods to `INavigationManager` would be nice.  However, we need to be thoughtful that adding members to an interface is a breaking change, since someone may be trying to provide an alternate implementation (at the very least via tests).